code/game/objects/effects/decals/cleanable.dm

```diff
@@ -11,6 +11,25 @@
 	plane = FLOOR_PLANE
 	///for blood n vomit in zero G --- IN GRAVITY=TRUE; NO GRAVITY=FALSE
 	var/gravity_check = TRUE
+	///prevents from calling clean_soon multiple times
+	var/cleaned_soon = FALSE
+
+/obj/effect/decal/cleanable/proc/_async_clean_soon(clean_time = 5 SECONDS)
+	var/iter = clean_time / (1 SECONDS)
+	for(var/i in 1 to iter)
+		if(QDELETED(src))
+			return
+		alpha -= (200 / iter)
+		if(alpha <= 55)
+			break
+		sleep(1 SECONDS)
+	qdel(src)
```

Comment:
This is terrible, I think there are 2 good alternatives here.
1. Make this use animate that fades, because I have a decent feeling that animate will be more performant than this, and then also use a QDEL_IN()
2. Port TG's cleaning bubbles effect, and have that displayed instead of the alpha changing, then use a QDEL_IN()

Reply:
replaced with animate() and QDEL_IN()